server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -156,6 +156,7 @@ public class TransportSearchAction extends HandledTransportAction<SearchRequest,
     private final NamedWriteableRegistry namedWriteableRegistry;
     private final CircuitBreaker circuitBreaker;
     private final SearchPipelineService searchPipelineService;
+    final List<SearchRequestOperationsListener> searchListenersList = new ArrayList<>();
```

Comment:
When/How do you intend to plug more listeners? Should we start with a single listener for now?

Reply:
@Bukhtawar Yes, we will be planning to add more listeners. Right now node stats is the only listener listening to these coordinator phase events and capturing it in indices stats. We also plan to add these as part of request level tracking and slow log on a request level as a whole. Each of these will be different listeners and we will use above logic to achieve it.

As mentioned [here](https://github.com/opensearch-project/OpenSearch/issues/7334)